main.go

```diff
@@ -64,24 +65,46 @@ func InfoHandler(contentAPI, needAPI, performanceAPI string, config *Config) fun
 			return
 		}
 
-		needStart := time.Now()
-		for _, needID := range artefact.Details.NeedIDs {
-			need, err := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
-			if err != nil {
-				renderError(w, http.StatusInternalServerError, "Need: "+err.Error())
+		var waitGroup sync.WaitGroup
+		errorChannel := make(chan string)
+
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+
+			needStart := time.Now()
+			for _, needID := range artefact.Details.NeedIDs {
+				need, nErr := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
+				if nErr != nil {
+					errorChannel <- "Need:" + nErr.Error()
```

Comment:
I'll make another pull request first, since error handling for the downstream
APIs could do with a tweak. 😀